### PR TITLE
fix: Fix csharp package name and Field equals/hashcode 

### PR DESCRIPTION
**Pull request by vam-google**
Also cleanup testing infrastructure to match current googleapis structure

### CODE REVIEW

File: google/cloud/compute/v1/BUILD.bazel [hunk 1]
```diff
@@ -3,64 +3,25 @@
...
     "//rules_gapic:disco_to_proto.bzl",
+    "gapic_yaml_from_disco",
+    "grpc_service_config_from_disco",
     "proto_from_disco",
 )
 
 proto_from_disco(
-    name = "compute_small",
-    src = "//:src/test/resources/compute.v1.small.json",
+    name = "compute",
```
Comment (vchudnov-g) [inline google/cloud/compute/v1/BUILD.bazel:12]: Would it be clearer to also call this `computesmall`? I had to look at the file to see the comment at the top that this is a demo file. The more descriptive name may help.

File: google/cloud/compute/v1small/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+]
+
+_SERVICE_IGNORE_LIST = [
+    "HttpHealthChecks",
+    "HttpsHealthChecks",
+]
+
+proto_from_disco(
+    name = "compute",
+    src = "//:src/test/resources/compute.v1.json",
```
Comment (vchudnov-g) [inline google/cloud/compute/v1small/BUILD.bazel:37]: Are your files swapped? This file is under `compute/v1small/` and references the full json, while the other one iunder`/v1/` refers to to `compute.v1small.json`

**Review by vchudnov-g — COMMENTED** [review]
The `proto_from_disco` `src`s appear to be swapped between compute.v1 and compute.v1small.

**Review by alexander-fenster — APPROVED** [review]
LGTM but something is wrong with the CI build.

File: google/cloud/compute/v1small/BUILD.bazel (hunk 2 above)
Comment (vam-google) [reply google/cloud/compute/v1small/BUILD.bazel:37]: oh, yeah, thanks! those files exist to simply run the generator without asserts (just a sanity check that the rules at least do not fail themselves. That is why even swapped version is "ok" but yes, it definitely should be cleaned up

**Pull request merged**